src/components/NewRoomModal.tsx

```diff
@@ -0,0 +1,167 @@
+			const hostname = window.location.hostname
+			const http = hostname === 'localhost' ? 'http' : 'https'
+			const port = window.location.port ? `:${window.location.port}` : ''
+			setDomain(`${http}://${hostname}${port}`)
+		}
+		const hostDataLocalStorage = localStorage.getItem('scrumDivingHostData')
+		if (hostDataLocalStorage) {
+			const hostData = JSON.parse(hostDataLocalStorage)
+			if (hostData?.hostName) {
+				setHostFirstName(hostData?.hostName)
+				setHideCreateBtn(false)
+				setShowNameInput(false)
+			}
+			console.log('%c>>> hostData LocalStorage', 'color: red', hostData)
+		}
+		setIsLoading(false)
+	}, [])
+
+	function handleOnChange(event: React.ChangeEvent<HTMLInputElement>) {
+		event.preventDefault()
+		setHostFirstName(event.target.value)
+		setHideCreateBtn(false)
+	}
+
+	function generateAlphaNumeric() {
+		return Math.random().toString(36).substring(3)
+	}
+
+	function copyToClipboard() {
+		if (roomUrlRef.current) {
+			navigator.clipboard.writeText(roomUrlRef.current.textContent || '')
+			alert('Room URL copied to clipboard!')
+		}
+		setHideGoBtn(false)
+	}
+
+	const handleShowModal = useCallback(() => {
+		dialogRef.current?.showModal()
+	}, [])
+
+	function handleCreateRoom() {
+		const newRoomNumber = generateAlphaNumeric() + generateAlphaNumeric()
+		setRoomUrl(`${domain}/${newRoomNumber}`)
+		setHostRoomUrl(`${domain}/host/${newRoomNumber}`)
+		if (hostRef.current) hostRef.current.value = ''
+		setHideGoBtn(true)
+		handleShowModal()
+		console.log('%c>>> hostFirstName:', 'color: red', hostFirstName)
+		console.log('%c>>> room:', 'color: #5f0', domain, newRoomNumber)
+	}
+
+	function goToNewRoom() {
+		console.log('%c>>> user room', 'color: red', roomUrl)
+		console.log('%c>>> host room', 'color: yellow', hostRoomUrl)
+		setHideCreateBtn(true)
+		const valuesLocalStorate = JSON.stringify({
+			hostName: hostFirstName,
+			roomUrl: hostRoomUrl,
+		})
+		localStorage.setItem('scrumDivingHostData', valuesLocalStorate)
+		router.push(hostRoomUrl)
+	}
+
+	return (
+		<div className='w-full flex flex-col items-center'>
+			{!isLoading && (
+				<div className='w-full flex flex-col items-center gap-6'>
+					{showNameInput ? (
+						<input
+							type='text'
+							ref={hostRef}
+							placeholder='Enter Your First Name'
+							onChange={handleOnChange}
+							className='input input-bordered input-info w-72 text-gray-200 placeholder:italic placeholder:text-info/90'
+						/>
+					) : (
+						<span className='text-2xl text-gray-300 tracking-wider pb-4'>
+							Welcome back {hostFirstName}!
+						</span>
+					)}
+
+					{/* <p className='text-sm text-gray-400 pt-2'>
+						Then click button to create a new room:
+					</p> */}
+
+					<button
+						type='button'
+						className='btn btn-warning btn-sm w-72 text-base'
+						onClick={handleCreateRoom}
+						disabled={hideCreateBtn}
+					>
+						Click to Create New Room
+					</button>
+				</div>
+			)}
+			<dialog ref={dialogRef} className='modal'>
```

Comment:
praise: Good usage of `dialog`! ⚡

Reply:
I've never used <dialog> before, but WOW, is it a billion times easier to make modals with it!! Thanks!

Reply:
Modal is gone. Reworked /host page.